arches/app/models/graph.py

```diff
@@ -1765,6 +1765,37 @@ def validate_fieldname(fieldname, fieldnames):
             if graphs_with_matching_slug.exists() and graphs_with_matching_slug[0].graphid != self.graphid:
                 raise GraphValidationError(_("Another resource model already uses the slug '{self.slug}'").format(**locals()), 1007)
 
+    def update_published_graphs(self):
+        """
+        Changes information in in GraphPublication models without creating
+        a new entry in graphs_x_published_graphs table
+        """
+        published_graphs = models.PublishedGraph.objects.filter(publication_id=self.publication_id)
+
+        for language_tuple in settings.LANGUAGES:
+            translation.activate(language=language_tuple[0])
+
+            serialized_graph = JSONDeserializer().deserialize(
+                JSONSerializer().serialize(self, force_recalculation=True)
+            )
+
+            published_graph_query = published_graphs.filter(language=language_tuple[0])
+            if not len(published_graph_query):
+                published_graph = models.PublishedGraph.objects.create(
+                    publication=self.publication,
+                    serialized_graph=serialized_graph,
+                    language=models.Language.objects.get(code=language_tuple[0]),
```

Comment:
This fails if a language is added to settings.py that doesn't exist in the Languages table.
This points out an issue.  The languages defined in settings.py are strictly for the UI.  Languages defined in the table are for business data (strings, concepts, etc...).  I'm guessing that the published graph should be using the langugages from `settings.py`